client/post-editor/editor-featured-image/preview-container.jsx

```diff
@@ -23,64 +22,33 @@ class EditorFeaturedImagePreviewContainer extends React.Component {
 		showEditIcon: PropTypes.bool,
 	};
 
-	state = {
-		image: null,
-	};
-
 	componentDidMount() {
-		this.fetchImage();
-		MediaStore.on( 'change', this.updateImageState );
+		this.props.fetchMediaItem( this.props.siteId, this.props.itemId );
 	}
 
 	componentDidUpdate( prevProps ) {
 		const { siteId, itemId } = this.props;
 		if ( siteId !== prevProps.siteId || itemId !== prevProps.itemId ) {
-			this.fetchImage();
+			this.props.fetchMediaItem( siteId, itemId );
+			this.props.onImageChange( itemId );
```

Comment:
Do we still want to check for whether the image is already being provided, as the old version did, before fetching?